src/mesh_tools.rs

```diff
@@ -0,0 +1,200 @@
+            Point3::new(0.0, 2.0, 1.0),
+        ];
+
+        let vertex_normals = vec![n(0.0, 0.0, 1.0)];
+
+        let faces = vec![
+            TriangleFace::new_separate(0, 3, 1, 0, 0, 0),
+            TriangleFace::new_separate(1, 3, 4, 0, 0, 0),
+            TriangleFace::new_separate(1, 4, 2, 0, 0, 0),
+            TriangleFace::new_separate(3, 5, 4, 0, 0, 0),
+            TriangleFace::new_separate(6, 7, 8, 0, 0, 0),
+        ];
+
+        Geometry::from_triangle_faces_with_vertices_and_normals(faces, vertices, vertex_normals)
+    }
+
+    fn triangular_island_geometry() -> Geometry {
+        let vertices = vec![
+            Point3::new(-1.0, 0.0, 1.0),
+            Point3::new(1.0, 0.0, 1.0),
+            Point3::new(0.0, 2.0, 1.0),
+        ];
+
+        let vertex_normals = vec![n(0.0, 0.0, 1.0)];
+
+        let faces = vec![TriangleFace::new_separate(0, 1, 2, 0, 0, 0)];
+
+        Geometry::from_triangle_faces_with_vertices_and_normals(faces, vertices, vertex_normals)
+    }
+
+    #[test]
+    fn test_separate_isolated_meshes_returns_identical_for_tessellated_triangle() {
+        let geometry = tessellated_triangle_geometry();
+
+        let calculated_geometries = separate_isolated_meshes(&geometry);
+
+        assert_eq!(calculated_geometries.len(), 1);
+
+        assert!(mesh_analysis::are_visually_identical(
+            &calculated_geometries[0],
+            &geometry
+        ));
+    }
+
+    #[test]
+    fn test_separate_isolated_meshes_returns_identical_for_cube() {
+        let geometry = geometry::cube_sharp_var_len([0.0, 0.0, 0.0], 1.0);
+
+        let calculated_geometries = separate_isolated_meshes(&geometry);
+
+        assert_eq!(calculated_geometries.len(), 1);
+        assert!(mesh_analysis::are_visually_identical(
+            &geometry,
+            &calculated_geometries[0]
+        ));
+    }
+
+    #[test]
+    fn test_separate_isolated_meshes_returns_identical_for_tessellated_triangle_with_island() {
+        let geometry = tessellated_triangle_with_island_geometry();
+        let geometry_triangle_correct = tessellated_triangle_geometry();
+        let geometry_island_correct = triangular_island_geometry();
+
+        let calculated_geometries = separate_isolated_meshes(&geometry);
+
+        assert_eq!(calculated_geometries.len(), 2);
+
+        if mesh_analysis::are_visually_identical(
+            &calculated_geometries[0],
+            &geometry_triangle_correct,
+        ) {
+            assert!(mesh_analysis::are_visually_identical(
+                &calculated_geometries[1],
+                &geometry_island_correct
+            ));
+        } else {
+            assert!(mesh_analysis::are_visually_identical(
+                &calculated_geometries[1],
+                &geometry_triangle_correct
+            ));
+            assert!(mesh_analysis::are_visually_identical(
+                &calculated_geometries[0],
+                &geometry_island_correct
+            ));
+        }
+    }
```

Comment:
it's not the same thing! it's comparing either 0<->0, 1<->1 or 0<->1, 1<->0

Reply:
I'm a bit lost now. What I wanted to say is that only the first branch of the condition ever runs since result of the first `mesh_analysis::are_visually_identical` is always the same.

Reply:
true, but at the beginning i didn't know which branch is actually executed. and i still don't know, tbh. i can try and figure and remove the other one but should i? the point here is that i don;t know which separate mesh comes first, so i'm ready for both cases.

Reply:
the important thing is that one is equal to one of the test samples and the other to the other one.

Reply:
Well, tests should be deterministic, so let's split this into 2 separate tests for each of the branches.

Reply:
okay, i'll do that

Reply:
actually, after many tests, it seems that the order of the two separate meshes changes each time. @yanchith mentioned some "seeds" used by HashSets - maybe that's why it changes each run?

Reply:
If I understand correctly, the test is deterministic with the branch, as it is now. I agree that it is a bit non-obvious, but it seems correct to me.